src/shared/components/Avatar/Avatar.tsx

```diff
@@ -1,40 +1,52 @@
 import React from 'react'
 import { CSSTransition } from 'react-transition-group'
-import { AvatarSize, Container, StyledImage, StyledTransitionGroup, StyledPlaceholder } from './Avatar.style'
+import {
+  AvatarSize,
+  Container,
+  StyledImage,
+  StyledTransitionGroup,
+  StyledPlaceholder,
+  SilhouetteAvatar,
+  EditButton,
+} from './Avatar.style'
 import { transitions } from '@/shared/theme'
+import Icon from '../Icon'
 
 export type AvatarProps = {
   onClick?: (e: React.MouseEvent<HTMLElement>) => void
-  handle?: string | null
+  onEditClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
   imageUrl?: string | null
   loading?: boolean
   className?: string
   size?: AvatarSize
+  editable?: boolean
 }
 
-const initialsFromHandle = (rawHandle: string | null | undefined): string => {
-  const handle = rawHandle || 'Unknown'
-  const vowels = ['a', 'e', 'i', 'o', 'u', 'y']
-  const [first = '', second = ''] = handle.split('')
-  return vowels.includes(second) ? first : `${first}${second}`
-}
-
-const Avatar: React.FC<AvatarProps> = ({ imageUrl, handle, loading = false, size = 'default', onClick, className }) => {
+const Avatar: React.FC<AvatarProps> = ({
+  imageUrl,
+  loading = false,
+  size = 'default',
+  onClick,
+  onEditClick,
+  className,
+  editable,
+}) => {
+  const isEditable = !loading && editable && size !== 'default'
   return (
     <Container onClick={onClick} size={size} className={className}>
       <StyledTransitionGroup>
+        {isEditable && (
+          <EditButton size={size} withAvatar={!!imageUrl} onClick={onEditClick}>
```

Comment:
Do you think `onEditClick` should stop event bubbling? Otherwise when I click the edit button, both handlers will be called